app/gui/src/dashboard/providers/ModalProvider.tsx

```diff
@@ -25,14 +33,94 @@ interface ModalContextType {
   readonly modal: Modal | null
 }
 
+const ModalsStore = createStore<{
+  readonly key: number
+  readonly modal: Modal | null
+  readonly setModal: (modal: ModalOrCallback | null) => void
+  readonly updateModal: (modal: ModalOrCallback) => void
+}>((set, get) => ({
+  key: 0,
```

Comment:
if all that matters is strict equality consider maybe using a symbol?

Reply:
What do you mean? Could you elaborate?

Key is just a number to make `Dialogs` remount when we change them (this prevents us from unexpected side effects). I don't think symbols are better here...

Reply:
ohh is this for a component? if so then never mind